pkg/addon/addon.go

```diff
@@ -56,6 +58,13 @@ func NewGetValuesFunc(imageName string) addonfactory.GetValuesFunc {
 		if err != nil {
 			return nil, err
 		}
+
+		// if cluster is hosed mode, the enableSyncLabelsToClusterClaims is false
```

Comment:
should we check the addon deploy mode or the klusterlet deploy mode?

Reply:
if the klusterlet is deployed in Hosted mode, but the work manager addon is deployed in Default mode(hypershift case in 2.5), will we enable or disable this functionality?

Reply:
You're right, I only considered SD case before, here should based on addon annotation.

Reply:
WIP

Reply:
Done.